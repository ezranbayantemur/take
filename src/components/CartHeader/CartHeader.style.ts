import {color, font, padding, typography} from '@style';
import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {},
  icon: {
    color: color.SMOKE,
    fontSize: typography.BIG,
    marginRight: padding.MEDIUM,
  },
  badge_container: {
    position: 'absolute',
    top: -8,
    right: -3,
    backgroundColor: color.MONZA,
    borderRadius: 50,
    width: 20,
    height: 20,
    justifyContent: 'center',
    alignItems: 'center',
  },
  badge_title: {
    color: color.ALABASTER,
    fontSize: typography.SMALL,
    fontFamily: font.BOLD,
  },
});
