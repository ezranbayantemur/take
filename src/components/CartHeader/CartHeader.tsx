import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {useSelector} from 'react-redux';
import {routes} from '@route';
import {RootState} from '../../redux/store';
import Icon from '../Icon/Icon';
import style from './CartHeader.style';

const CartHeader = () => {
  const navigation = useNavigation<any>();
  const productInCartCount = useSelector<RootState, number>(
    state => state.cart.allProductsInCart.length,
  );

  const isShowBadge = productInCartCount > 0;

  const handleOnPress = () => {
    navigation.navigate(routes.CART);
  };

  return (
    <TouchableOpacity testID="cartheader_touchable" onPress={handleOnPress}>
      <Icon name="cart" style={style.icon} />
      {isShowBadge && (
        <View style={style.badge_container}>
          <Text testID="cartheader_touchable_text" style={style.badge_title}>
            {productInCartCount}
          </Text>
        </View>
      )}
    </TouchableOpacity>
  );
};

export default CartHeader;
