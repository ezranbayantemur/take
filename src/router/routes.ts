enum routes {
  LOGIN = 'LoginScreen',
  REGISTER = 'RegisterScreen',
  DISCOVER = 'DiscoverScreen',
  PRODUCTS = 'ProductsScreen',
  PRODUCT_DETAIL = 'ProductDetailScreen',
  CART = 'CartScreen',
}

export default routes;
