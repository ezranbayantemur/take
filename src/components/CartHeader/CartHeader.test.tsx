import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import CartHeader from './CartHeader';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import routes from '@route';

const mockOnProductSelect = jest.fn();
const mockStore = configureMockStore();
const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      navigate: mockNavigate,
    }),
  };
});

const store = mockStore({
  cart: {
    productsInCart: [
      {
        id: 0,
      },
      {
        id: 1,
      },
    ],
  },
});

let wrapper: ReturnType<typeof render>;

describe('CategoryCard unit tests', () => {
  beforeEach(() => {
    wrapper = render(
      <Provider store={store}>
        <CartHeader />
      </Provider>,
    );
    mockOnProductSelect.mockClear();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render product count correctly', () => {
    expect(wrapper.queryByText('2')).not.toBeNull();
  });

  it('should navigate to cart page on press', () => {
    const touchable = wrapper.getByTestId('cartheader_touchable');
    fireEvent(touchable, 'onPress');

    expect(mockNavigate).toBeCalledWith(routes.CART);
  });
});
