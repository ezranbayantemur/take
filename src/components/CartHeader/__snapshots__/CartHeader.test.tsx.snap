// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartCard unit tests should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="cartheader_touchable"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 12,
        },
        {
          "color": "#BDBDBD",
          "fontSize": 20,
          "marginRight": 16,
        },
        {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
    suppressHighlighting={true}
  >
    î¤º
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#DB040E",
        "borderRadius": 50,
        "height": 20,
        "justifyContent": "center",
        "position": "absolute",
        "right": -3,
        "top": -8,
        "width": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FBFBFB",
          "fontFamily": "Lato-Bold",
          "fontSize": 13,
        }
      }
    >
      2
    </Text>
  </View>
</View>
`;
